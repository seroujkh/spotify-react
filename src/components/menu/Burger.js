

const Burger = props => {

    return (
        <div className= {"menu-btn" +( props.open ?  ' open ' : '')} onClick={props.toggleClick} >
            <span className={"menu-btn__burger" +( props.open ?  ' open ' : '')}></span>
        </div>
    )


}

export default Burger;