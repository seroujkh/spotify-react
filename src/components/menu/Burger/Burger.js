const Burger = props => {

    return (
        <div data-testId="burger-1" className= {"menu-btn" +( props.open ?  ' open ' : '')} onClick={props.toggleClick} >
            <span  data-testId="burger-2"  className={"menu-btn__burger" +( props.open ?  ' open ' : '')}></span>
        </div>
    )
}

export default Burger;