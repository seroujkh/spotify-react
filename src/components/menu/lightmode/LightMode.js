import React, { useEffect, useState, useContext } from "react";
import AuthContext from "../../../store/auth-context";
import PropTypes from 'prop-types';

const LightMode = (props) => {

    const [lightMode, setMode] = useState(props.mode);
    const ctx = useContext(AuthContext);


    useEffect(() => {
        setMode(props.mode);
    }, [props.mode]);

    const toggleLightModeHandler = () => {
        if (lightMode === 'light') {
            setMode("dark");
            ctx.setLightMode("dark");
        } else {
            ctx.setLightMode("light");
            setMode("light");
        }
    }

    return (
        <div className="lightmode d-flex justify-content-between px-3 mt-4" onClick={toggleLightModeHandler}>
            <span className="text-white">Light</span>
            <div className="toggle_light ">
                <div className={(lightMode === 'dark' ? ' right ' : ' left ')}></div>
            </div>
            <span className="text-white">Dark</span>
            <div className="active d-none"  data-testid="active">{lightMode}</div>
        </div>
    )
}

LightMode.propTypes = {
    mode: PropTypes.string.isRequired
}
export default LightMode;