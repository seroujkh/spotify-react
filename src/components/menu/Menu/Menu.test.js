import { render, screen } from '@testing-library/react';
import SidebarMenu from './Menu';
import { shallow, mount } from 'enzyme';
import React from 'react';
import LightMode from '../lightmode/LightMode';
import Burger from '../Burger';

describe('Menu', () => {
    let wrapper;

    beforeEach(() => wrapper = shallow(<SidebarMenu />));

    it('should render correctly', () => expect(wrapper).toMatchSnapshot());

    it('should render the LightMode Component', () => {
        expect(wrapper.containsMatchingElement(<LightMode mode="light" />)).toEqual(true);
    });
    it('should render the Burger Component', () => {
        expect(wrapper.containsMatchingElement(<Burger />)).toEqual(true);
    });


});