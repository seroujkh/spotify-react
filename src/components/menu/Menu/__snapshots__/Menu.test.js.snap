// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render correctly 1`] = `ShallowWrapper {}`;
